lib/Net/Async/Blockchain/ETH.pm

```diff
@@ -433,127 +451,21 @@ hashref from an array of L<Net::Async::Blockchain::Transaction>
 
 =cut
 
-async sub _check_contract_transaction {
-    my ($self, $transaction) = @_;
-
-    # ERC20 transfer
-    # transfer(address _to, uint256 _value)
-    # signature = 4 bytes
-    # address = 32 bytes
-    # amount = 32 bytes
-    # total = 68 bytes, characters = 136
-    # including "0x" = 138
-    if (length($transaction->data) >= 138 && substr($transaction->data, 0, 10) eq substr(TRANSFER_SIGNATURE, 0, 10)) {
-        my $address = substr($transaction->data, 10, 64);
-        my $amount  = substr($transaction->data, 74, 64);
-
-        return undef unless $address && $amount;
-
-        my $contract_address = $transaction->to;
-
-        my $hex_symbol = await $self->rpc_client->call({
-                data => SYMBOL_SIGNATURE,
-                to   => $contract_address
-            },
-            "latest"
-        );
+async sub _check_plugins {
+    my ($self, $transaction, $receipt) = @_;
 
-        my $symbol = $self->_to_string($hex_symbol);
-        return undef unless $symbol;
+    my @modules = $filter->modules();
```

Comment:
Could these be filtered prior to entering this method?

Reply:
the `modules` method from `Module::PluginFinder`  does not check the filter https://metacpan.org/release/Module-PluginFinder/source/lib/Module/PluginFinder.pm#L163-167

Reply:
+1. The filtering will be quite slow if you call `->modules` every time here. Additionally since it doesn't look like you're even using the filtering/find-one behaviour of `Module::PluginFinder` maybe you just wanted to use `Module::Pluggable`. But additionally it would be best not to mingle that change with the memory leak fix.